backend/server/controllers/user.js

```diff
@@ -466,3 +466,104 @@ export const getFollow = async (req, res) => {
         res.status(500).send({ 'Error message': error.toString() })
     }
 }
+
+export const modifyUserByUsername = async (req, res) => {
+    try {
+        const { params } = req
+        const userOfInterest = await models.users.findOne({
+            where: {
+                username: params.username
+            }
+        })
+
+        const authToken = req.get('Authorization')
+
+        if (!authToken) {
+            res.status(400).send({
+                'Error message': 'Auth token not provided',
+            })
+        }
+
+        const loggedInUser = Authentication.extractUser(authToken)
+
+        if (!loggedInUser) {
+            res.status(401).send({
+                'Error message': 'Auth token invalid',
+            })
+            return
+        } else {
+
+            const { params, body} = req
+
+            if (!userOfInterest) {
+                res.status(404).send('Invalid username.')
+
+            } else {
+
+                const updatedUser = await models.userOfInterest.update({
```

Comment:
User also has some fields such as bio and profilePic that we may want to update.

Reply:
Just adding to this, the only fields that can't be edited are the followers and the join date